packages/core/test/invoke.test.ts

```diff
@@ -2614,8 +2637,9 @@ describe('invoke', () => {
       expect(actorStarted).toBe(false);
     });
 
+    // TODO: determine the correct behavior for this: start then stop, or never start (more complicated)?
```

Comment:
It would be at least good to determine the expected behavior for this and either remove this test or create a follow-up ticket to implement the required changes. 

I don't mind not implementing this behavior but we'd deviate from the SCXML spec by doing so. This behavior is a small optimization at best, OTOH if we could implement it and save some resources on behalf of our users automatically - maybe we should.

If we plan to remove this behavior then we should also just assign this unconditionally because the condition becomes redundant without this expectation:
https://github.com/statelyai/xstate/blob/1453c370e986848c6f8b40b26e163d7881599c12/packages/core/src/exec.ts#L116-L120

Reply:
I found a way to get it working; just need to fix a type.